### PR TITLE
pkg/openthread: deprecate ot_command related functions

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
Since #15038 saw the light, the current "ot_command" wrapper is only limiting OpenThread usage. The Event Queue makes it possible now to call OpenThread API functions directly when posting an event to the OpenThread queue.

This PR removes the `ot_command` calls from `examples/openthread`, improve Doxygen documentation and deprecate the "ot_command" related functions

### Testing procedure
Test if `examples/openthread` still works ok. Note that it's possible to run OpenThread with any board that supports the Radio HAL if using `netdev_ieee802154_submac`.

### Issues/PRs references
None so far

### CODE REVIEW

File: examples/openthread/main.c [hunk 1]
```diff
@@ -16,15 +16,37 @@
...
 
 int main(void)
 {
-    puts("This a test for OpenThread");
     /* Example of how to call OpenThread stack functions */
-    puts("Get PANID ");
-    uint16_t panid = 0;
-    uint8_t res = ot_call_command("panid", NULL, (void*)&panid);
-    printf("Current panid: 0x%x (res:%x)\n", panid, res);
+    puts("This a test for OpenThread");
```
Comment (aabadie) [inline examples/openthread/main.c:42]: openthread is in examples/ so not really a test. I know it was there already but maybe we can take the opportunity of this PR to adapt.

Comment (jia200x) [reply examples/openthread/main.c:42]: Sure, I can do it

Comment (jia200x) [reply examples/openthread/main.c:42]: done!

**Comment by jia200x** [comment]
ping @aabadie

**Comment by jia200x** [comment]
let's postpone this one

**Comment by jia200x** [comment]
ping

**Review by benpicco — APPROVED** [review]
Please squash!

**Comment by MrKevinWeiss** [comment]
Tested and it looks good:

```
Welcome to pyterm!
Type '/exit' to exit.
version
2021-06-14 15:57:15,777 # version
2021-06-14 15:57:15,783 # OPENTHREAD/thread-reference-20180926; NONE; Jun 14 2021 15:54:31
2021-06-14 15:57:15,784 # Done
> panid
2021-06-14 15:57:17,977 #  panid
2021-06-14 15:57:17,978 # beef
2021-06-14 15:57:17,978 # Done
> 
```

**Comment by MrKevinWeiss** [comment]
please rebase to maybe fix the pr_check error...

**Comment by jia200x** [comment]
thanks for the review!

**Pull request merged**